src/index.ts

```diff
@@ -197,7 +200,9 @@ export function createServer(
     }
   }
 
-  const server = http.createServer(async (req, res) => {
+  const httpModule: any = serverOptions.https ? https : http;
+  const httpOptions: any = httpModule === https ? serverOptions.https : serverOptions.http;
+  const server = httpModule.createServer(httpOptions ?? {}, async (req: IncomingMessage, res: ServerResponse) => {
```

Comment:
Could we rewrite this in a way where we would get type safety? Something like this: 
```ts
server = (serverOptions.https ? https.createServer(serverOptions.https) : http(serverOptions.http))
```

Reply:
thanks, I've updated it